### PR TITLE
feat: add support for events

### PR DESCRIPTION
**Pull request by Ernst79**
- Add support for events, for dimmers, switches and buttons
- Reordered the `SensorLibrary` to an alphabetical order

### CODE REVIEW

File: src/sensor_state_data/device_trigger/event_device_keys.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Supported devices keys for devices that fire events
+
+from ..enum import StrEnum
+
+
+class EventDeviceKeys(StrEnum):
```
Comment (balloob) [inline src/sensor_state_data/device_trigger/event_device_keys.py:6]: Is there a reason why we would need to go this far with defining a data model? Why can't we have the integration just listen for bluetooth packets that are marked as "event" by the sensor-state-data ?

File: src/sensor_state_data/device_trigger/event_device_keys.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Supported devices keys for devices that fire events
```
[the remark this answers is not in the document]
Comment (Ernst79) [reply src/sensor_state_data/device_trigger/event_device_keys.py]: the entire folder has become obsolete with your second comment, so it is removed now

File: src/sensor_state_data/device_trigger/event_device_keys.py (hunk 1 above)
Comment (Ernst79) [reply src/sensor_state_data/device_trigger/event_device_keys.py]: Removed it, also the event_types file, for the same reason.

File: src/sensor_state_data/data.py [hunk 3]
```diff
@@ -279,3 +285,27 @@ def update_sensor(
             native_unit_of_measurement=native_unit_of_measurement,
             device_class=device_class,
         )
+
+    def update_event(
```
Comment (balloob) [inline src/sensor_state_data/data.py:289]: Should the name be `fire_event`, like the doc string says it does?

File: src/sensor_state_data/library.py [hunk 4]
```diff
@@ -44,31 +36,21 @@ class SensorLibrary:
...
-        device_class=SensorDeviceClass.PM25,
-        native_unit_of_measurement=Units.CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    KEG_SIZE__VOLUME_LITERS = BaseSensorDescription(
+        device_class=SensorDeviceClass.KEG_SIZE,
+        native_unit_of_measurement=Units.VOLUME_LITERS,
     )
-    CO2__CONCENTRATION_PARTS_PER_MILLION = BaseSensorDescription(
-        device_class=SensorDeviceClass.CO2,
-        native_unit_of_measurement=Units.CONCENTRATION_PARTS_PER_MILLION,
+    KEG_TYPE__NONE = BaseSensorDescription(
```
Comment (balloob) [inline src/sensor_state_data/library.py:47]: Are these unrelated changes?

**Review by balloob — APPROVED** [review]

File: src/sensor_state_data/library.py (hunk 4 above)
Comment (Ernst79) [reply src/sensor_state_data/library.py:47]: This is just a reordering of this list in alphabetical order, See the first post

File: src/sensor_state_data/data.py (hunk 3 above)
Comment (Ernst79) [reply src/sensor_state_data/data.py]: yes, sound good to me. Changed it.

File: src/sensor_state_data/value.py [hunk 5]
```diff
@@ -26,3 +26,11 @@ class BinarySensorValue(BaseValue):
...
 
     native_value: bool | None
+
+
+@dataclass(frozen=True)
+class Event(BaseValue):
+    """A class that describes device events."""
+
+    event_type: str | int | float | date | datetime | Decimal | None
+    event_subtype: str | int | float | date | datetime | Decimal | None
```
Comment (bdraco) [inline src/sensor_state_data/value.py:36]: Can the event_type really be `None`?

File: src/sensor_state_data/data.py [hunk 6]
```diff
@@ -279,3 +285,27 @@ def update_sensor(
...
+        key: str,
+        event_type: None | str | int | float | date | datetime | Decimal,
+        event_subtype: None
+        | str
+        | int
+        | float
+        | date
+        | datetime
+        | Decimal
+        | None = None,
```
Comment (bdraco) [inline src/sensor_state_data/data.py:300]: I think it make sense to only allow `str` for `event_type` and `str | None = None` for event_subtype

**Review by bdraco — CHANGES_REQUESTED** [review]
Some typing comments above

File: src/sensor_state_data/data.py (hunk 6 above)
Comment (Ernst79) [reply src/sensor_state_data/data.py]: Changed

File: src/sensor_state_data/value.py (hunk 5 above)
Comment (Ernst79) [reply src/sensor_state_data/value.py:36]: No, it should not. Changed it.

**Review by bdraco — APPROVED** [review]

**Review by bdraco — APPROVED** [review]
